test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,156 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var KumaSDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid KumaSDConfig - TLS with Secrets",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server: "http://kuma-control-plane.example.com",
+					TLSConfig: &monitoringv1.SafeTLSConfig{
+						CA: monitoringv1.SecretOrConfigMap{
+							Secret: &v1.SecretKeySelector{
+								Key: "ca-cert",
```

Comment:
Also the pipeline issues seems to be related to indentation, so `"ca-cert"` needs to be at the same level of `v1.LocalObjectReference{Name: "kuma-tls-secret"},`, please check:
https://github.com/prometheus-operator/prometheus-operator/actions/runs/12506100526/job/34890510497?pr=7241